pkg/memorystore/memorystore.go

```diff
@@ -80,11 +80,28 @@ func addParentsAndChildren(parent, path []string, secretConfig *types.SecretConf
 					if Equal(node.Path, path) {
 						node.Parents = append(node.Parents, parentNode)
 						parentNode.Children = append(parentNode.Children, node)
+						break
 					}
 				}
 			}
 		}
 	}
+	// all aliases should be parents of the relevant secret key
+	for _, node := range nodes {
```

Comment:
does this path get tested at all?

Reply:
Good question, and yes it does. That's what's in the pkg/memorystore/test/helpers.go changes. I test drove the changes, first updating the tests, then updating the code to make the tests pass.